modules/internal/ArrayViewRankChange.chpl

```diff
@@ -436,6 +436,7 @@ module ArrayViewRankChange {
   // other array class implementations, it supports the standard dsi
   // interface.
   //
+  pragma "aliasing array"
```

Comment:
The need for this pragma makes me nervous, though I'm not sure why. @benharsh any objections to needing to mark that array views can create aliases?

Reply:
I think this would be OK for now, but I wonder if it would be better to assume that an array *does* alias by default and require that users opt-in for non-aliasing. Technically there's nothing stopping today's users from creating aliasing array distributions, as far as I know.

Reply:
It'd be fine with me to reverse the pragma, but the core concept of this change relies on the fact that different _array records don't have instances or elements that alias each other unless they are array views. (Which wasn't always true, but I've been quietly building up to it for a couple years now :)